test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -95,3 +525,147 @@ func cpuSpecToString(cpus *performancev2.CPU) string {
 	}
 	return sb.String()
 }
+
+// deleteTestPod removes guaranteed pod
+func deleteTestPod(testpod *corev1.Pod) {
+	// it possible that the pod already was deleted as part of the test, in this case we want to skip teardown
+	err := testclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(testpod), testpod)
+	if errors.IsNotFound(err) {
+		return
+	}
+
+	err = testclient.Client.Delete(context.TODO(), testpod)
```

Comment:
Please pass the `context` from outside

Reply:
Addressed in the latest commit

Reply:
you are creating new `context` here. you should pass `ctx` (the one on line 474)

Reply:
Addressed in the latest commit